crypto/o_time.c

```diff
@@ -358,3 +358,19 @@ static void julian_to_date(long jd, int *y, int *m, int *d)
     *m = j + 2 - (12 * L);
     *y = 100 * (n - 49) + i + L;
 }
+
+
+int OPENSSL_timegm(struct tm *tm, time_t *t)
+{
+    /* Figures it out without mucking the environment */
+    long basetime, currtime;
+
+    basetime = date_to_julian(1970, 1, 1);
+    currtime = date_to_julian(tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday);
+
+    if (t != NULL)
+        *t = (((currtime - basetime) * 24 + tm->tm_hour) * 60 + tm->tm_min) * 60 + tm->tm_sec;
```

Comment:
> So, if your system can only deal with 32-bit time_t, then you will have a problem handling those dates regardless of this function.

No that is not true, I can perfectly validate a certificate that is valid until 2099 on my
linux-32 *because* that does not involve converting anything back to time_t.
Moreover I can even generate this certificate on linux-32, because ASN_TIME_adj
only needs the current time and can easily add 36500 days to it without using time_t
arithmentics  for that, time_t arithmentics are not a really good idea.